### PR TITLE
Fix: Cannot mark as impressed more than one message almost same time

### PR DESCRIPTION
**Pull request by eldhosembabu**
https://github.com/firebase/firebase-android-sdk/issues/3774

The Flag variable to check whether impressed or not was a static one and because of which when one FIAM message is marked as impressed, it was affecting other messages until the app restarts.

### CODE REVIEW

**Comment by vkryachko** [comment]
@eldhosembabu can you add tests for this change?

**Comment by eldhosembabu** [comment]
> @eldhosembabu can you add tests for this change?

Done

File: firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/DisplayCallbacksImpl.java [hunk 1]
```diff
@@ -72,6 +72,11 @@ public class DisplayCallbacksImpl implements FirebaseInAppMessagingDisplayCallba
     wasImpressed = false;
   }
 
+  @VisibleForTesting
+  boolean hasImpressed() {
+    return wasImpressed;
```
Comment (vkryachko) [inline firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/DisplayCallbacksImpl.java:77]: curious why the method is called `hasImpressed`, while it accesses the variable called `wasImpressed`.

Comment (eldhosembabu) [reply firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/DisplayCallbacksImpl.java]: Changed the method name to wasImpressed. This is introduced just for writing unit test.

**Review by vkryachko — APPROVED** [review]

**Pull request merged**